clients/src/main/java/org/apache/kafka/common/requests/PushTelemetryRequest.java

```diff
@@ -60,17 +62,47 @@ public PushTelemetryRequest(PushTelemetryRequestData data, short version) {
 
     @Override
     public PushTelemetryResponse getErrorResponse(int throttleTimeMs, Throwable e) {
-        PushTelemetryResponseData responseData = new PushTelemetryResponseData()
-                .setErrorCode(Errors.forException(e).code())
-                .setThrottleTimeMs(throttleTimeMs);
-        return new PushTelemetryResponse(responseData);
+        return errorResponse(throttleTimeMs, Errors.forException(e));
     }
 
     @Override
     public PushTelemetryRequestData data() {
         return data;
     }
 
+    public PushTelemetryResponse errorResponse(int throttleTimeMs, Errors errors) {
+        PushTelemetryResponseData responseData = new PushTelemetryResponseData();
+        responseData.setErrorCode(errors.code());
+        /*
+         THROTTLING_QUOTA_EXCEEDED is thrown in telemetry APIs when the telemetry request
+         arrives prior the configured minimum interval between two consecutive telemetry requests.
+         In this case, the throttleTimeMs should not be included in the response to avoid the
+         muting of the client channel for all the other requests.
+        */
+        if (Errors.THROTTLING_QUOTA_EXCEEDED != errors) {
```

Comment:
We shouldn't set throttleTimeMs even for Errors.THROTTLING_QUOTA_EXCEEDED. throttleTimeMs can only be set when there is no error.

Reply:
I have removed the conditional llogic in this class and set `0` as the throttle time in case of error in ClientMetricsManager. Done.